### PR TITLE
[DropdownMenu][ContextMenu] Provide explicit `Sub` parts

### PR DESCRIPTION
**Pull request by andy-hook**
Closes https://github.com/radix-ui/primitives/issues/1309

- [x] https://github.com/radix-ui/primitives/pull/1393
- [x] https://github.com/radix-ui/primitives/pull/1396
- [x] https://github.com/radix-ui/primitives/pull/1406

### CODE REVIEW

**Comment by benoitgrelard** [comment]
When doing stacked like this I think we need to do the version only in the top-level PR as otherwise it produces multiple version files and it's not happy about it.

**Comment by andy-hook** [comment]
> When doing stacked like this I think we need to do the version only in the top-level PR as otherwise it produces multiple version files and it's not happy about it.

Indeed, i'll fix it when all the changes are in.

**Comment by benoitgrelard** [comment]
Playing with the stories, I noticed that the Menu>submenus story is broken:
CleanShot 2022-05-20 at 10 15 17@2x

File: packages/react/context-menu/src/ContextMenu.tsx [hunk 1]
```diff
@@ -134,15 +120,17 @@ const ContextMenuTrigger = React.forwardRef<ContextMenuTriggerElement, ContextMe
...
             handleOpen(event);
+            event.preventDefault();
           })}
           onPointerDown={composeEventHandlers(
             props.onPointerDown,
             whenTouchOrPen((event) => {
               // clear the long press here in case there's multiple touch points
               clearLongPress();
-              longPressTimerRef.current = window.setTimeout(() => handleOpen(event), 700);
+              longPressTimerRef.current = window.setTimeout(() => {
```
Comment (benoitgrelard) [inline packages/react/context-menu/src/ContextMenu.tsx:131]: nit: was there a reason for this change?

**Review by benoitgrelard — APPROVED** [review]
All looks good for me apart from that broken story.

File: packages/react/context-menu/src/ContextMenu.tsx (hunk 1 above)
Comment (andy-hook) [reply packages/react/context-menu/src/ContextMenu.tsx]: Formatting hangover from a previous change I think

**Pull request merged**